icicle/CMakeLists.txt

```diff
@@ -37,7 +37,9 @@ set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 # Build options
 option(BUILD_TESTS "Build unit tests. Default=OFF" OFF)
 option(CPU_BACKEND "Build CPU backend. Default=ON" ON)
+# TODO Yuval: consider decoupling backends from frontend build. Is that worth the effort?
```

Comment:
is it?
Over time the cpu backend compilation time will be less efficient

Reply:
I referred to cuda and metal. I did not intend to split the CPU from this repo or build it separately.
We can discuss it but it's not related to this PR.
About metal and cuda, we could separate them entirely from the frontend cmake but it will complicate the backend. I am not sure if it is really worth it at this point although it is probably better to decouple the backends build entirely if we can.